### PR TITLE
[webgpu] Fix that exceeding limitation errors

### PR DESCRIPTION
**Pull request by qjia7**
Recently, webgpu adds more strict limitations for the work group size
and shared memory size. This PR fixes those errors caused by exceeding
limitations. It may hurt performance under some situations. The fine
tuning will be added after we have enough experiments.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5385)

### CODE REVIEW

**Comment by qjia7** [comment]
@kainino0x @lina128 @jinjingforever Please take a look, thanks.

cc @haoyunfeix @axinging

**Review by lina128 — APPROVED** [review]
LGTM from TFJS, please wait for @kainino0x review before merge.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5385#-:-Mfcrm9g1IBro4dZbLWA:b-z2yrci)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @jinjingforever and @kainino0x)

**Review by kainino0x — APPROVED** [review]

**Pull request merged**
